Makefile.common

```diff
@@ -33,10 +35,15 @@ endif
 
 # Check that rustc is the right nightly - warn not error if wrong, sometimes useful to test others
 RUSTC_VERSION_STRING := rustc 1.16.0-nightly (83c2d9523 2017-01-24)
-ifneq ($(shell rustc --version),$(RUSTC_VERSION_STRING))
-    $(warning Unexpected rustc version. Expected $(RUSTC_VERSION_STRING) got $(shell rustc --version))
+ifeq ($(shell rustup -V &> /dev/null && echo success),success)
+  DUMMY := $(shell rustup install $(RUSTUP_TOOLCHAIN))
```

Comment:
Oic... how about this:

the logic is:

  1. check to see if the expected version of `rustc` is installed (we don't care how, on the path? cool. from rustup? cool).
  2. if not, just blindly try installing with `rustup`.
  3. check again
  4. if after all that, it's not installed, spit out the warning.

It involves the same check twice, but meh...

```makefile
RUSTC_VERSION_STRING := rustc 1.16.0-nightly (83c2d9523 2017-01-24)
ifneq ($(shell RUST_TOOLCHAIN=$(RUSTUP_TOOLCHAIN) rustc --version), $(RUSTC_VERSION_STRING))
  DUMMY := $(shell rustup install $(RUSTUP_TOOLCHAIN))
  ifneq ($(shell RUST_TOOLCHAIN=$(RUSTUP_TOOLCHAIN) rustc --version), $(RUSTC_VERSION_STRING))
    $(warning You do not have the correct version of `rustc` configured, and `rustup` is not installed.)
    $(warning Expected `rustc` version "$(RUSTC_VERSION_STRING)" but got "$(shell rustc --version)")
    $(warning You may experience unexpected compilation warnings or errors)
    $(warning To fix, install `rustup` from https://rustup.rs/, then run: `rustup install $(RUSTUP_TOOLCHAIN)`)
  endif
endif
```